app/src/main/java/org/mozilla/fenix/components/metrics/SecurePrefsTelemetry.kt

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import android.content.Context
+import android.os.Build
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import mozilla.components.lib.dataprotect.SecurePrefsReliabilityExperiment
+import mozilla.components.service.nimbus.NimbusApi
+import org.mozilla.fenix.experiments.ExperimentBranch
+import org.mozilla.fenix.experiments.Experiments
+import org.mozilla.fenix.ext.withExperiment
+
+/**
+ * Allows starting a quick test of ACs SecureAbove22Preferences that will emit Facts
+ * for the basic operations and allow us to log them for later evaluation of APIs stability.
+ */
+class SecurePrefsTelemetry(
```

Comment:
If it's okay with you I would prefer keeping `SecurePrefsTelemetry(..).startTests()` just because of the better readability and having a cleaner separation from the rest of the code.